src/profile-logic/symbol-store.js

```diff
@@ -278,7 +278,16 @@ export class SymbolStore {
     // database. This is delayed until after the request has been kicked off
     // because it can take some time.
     // We also need a demangling function for this, which is in an async module.
-    const demangleCallback = (await demangleModulePromise).demangle_any;
+    // Module loading can fail (for example in browsers without WebAssembly
+    // support, or due to bad server configuration), so we will fall back to a
+    // pass through function if that happens.
+    const demangleCallback = await demangleModulePromise.then(
+      demangleModule => demangleModule.demangle_any,
+      error => {
+        console.error('Demangling module could not be imported.', error);
+        return mangledString => mangledString;
+      }
+    );
```

Comment:
I had considered this solution but decided against it.
I agree that the mix of then and await is a bit awkward, but I find the pattern of declaring a variable first and then assigning it in two different places even more awkward, because it makes it less clear how the values flow. That's a matter of personal taste though.
I've kept my approach of using both then and await, but I've followed your suggestion of moving the new comments into the error handling branch.